lerobot/common/datasets/utils.py

```diff
@@ -80,7 +82,28 @@ def hf_transform_to_torch(items_dict: dict[torch.Tensor | None]):
     return items_dict
 
 
-def load_hf_dataset(repo_id, version, root, split) -> datasets.Dataset:
+@cache
```

Comment:
FYI I think we should avoid using `cache`. It's not super common so it's easy to miss and as a result makes things hard to read.

I am ok to leave it for this PR, as the code is already working well and `cache` was used wisely to minimize the amount of code changes.